distributed/core.py

```diff
@@ -484,7 +487,9 @@ def handle_stream(self, comm, extra=None, every_cycle=[]):
                 pdb.set_trace()
             raise
         finally:
-            comm.close()  # TODO: why do we need this now?
+            yield comm.close()  # TODO: why do we need this now?
+            #             ^ Good question :) comm.close can be a
+            # coroutine, in which case this isn't doing anything.
```

Comment:
Not true actually, tornado coroutines are automatically run in the background, even if they are not awaited/yielded.